### ISSUE TITLE
Status 500 while importing issues

### ISSUE THREAD

**Issue by ooxi** [trentm/googlecode2github#5]
Hi,

while importing the issues from [Google Code](http://code.google.com/p/violetland/issues) to [Github](https://github.com/ooxi/violetland) I experienced Github returning status 500 without a message for the following issues
- [Issue 5](http://code.google.com/p/violetland/issues/detail?id=5)
- [Issue 15](http://code.google.com/p/violetland/issues/detail?id=15)
- [Issue 24](http://code.google.com/p/violetland/issues/detail?id=24)
- [Issue 29](http://code.google.com/p/violetland/issues/detail?id=29)
- [Issue 33](http://code.google.com/p/violetland/issues/detail?id=33)

I could not find out why this happens. I doesn have something todo with the orignal content, because if i replace it in those tickes, they are getting imported. I tried `body.encode('ascii', 'replace')` instead of `body.encode('utf-8')` but that didn't help.

-- [ooxi](https://github.com/ooxi)

**Comment by trentm** [comment]
I can repro:

```
Migrating issue 5.
     from: http://code.google.com/p/violetland/issues/detail?id=5
       to: https://github.com/trentm/ghtest/issues/5
-- Google code issue
{'author': {'name': 'NFLaZ...@gmail.com', 'uri': '/u/108645236195463941083/'},
 'content': u'[  8%] Building CXX object CMakeFiles/violetland.dir/src/game/\nWeapon.cpp.o                                                    \n/var/abs/local/yaourtbuild/violetland-svn/src/violetland/src/game/Weapon.cpp: In member \nfunction \u2018std::vector<Bullet*, std::allocator<Bullet*> >* Weapon::fire(float, \nfloat)\u2019:                                                                              \n/var/abs/local/yaourtbuild/violetland-svn/src/violetland/src/game/Weapon.cpp:55: \u043e\u0448\u0438\u0431\u043a\u0430: \n\u2018Bullet::BulletType\u2019 is not a class or \nnamespace                                                                                                                   \n/var/abs/local/yaourtbuild/violetland-svn/src/violetland/src/game/Weapon.cpp:58: \u043e\u0448\u0438\u0431\u043a\u0430: \n\u2018Bullet::BulletType\u2019 is not a class or \nnamespace                                                                                                                   \nmake[2]: *** [CMakeFiles/violetland.dir/src/game/Weapon.cpp.o] \u041e\u0448\u0438\u0431\u043a\u0430 \n1                                                       \nmake[1]: *** [CMakeFiles/violetland.dir/all] \u041e\u0448\u0438\u0431\u043a\u0430 2                                                                         \nmake: *** [all] \u041e\u0448\u0438\u0431\u043a\u0430 2                                                                                                      \n==> \u041e\u0428\u0418\u0411\u041a\u0410: \u0421\u0431\u043e\u0440\u043a\u0430 \u0437\u0430\u0432\u0435\u0440\u0448\u0438\u043b\u0430\u0441\u044c \u043d\u0435\u0443\u0434\u0430\u0447\u043d\u043e.                                                                                      \n    \u041f\u0440\u0435\u0436\u0434\u0435\u0432\u0440\u0435\u043c\u0435\u043d\u043d\u044b\u0439 \u0432\u044b\u0445\u043e\u0434...                                                                                                  \nError: Makepkg was unable to build violetland-svn package.',
 'id': '5',
 'labels': ['Type-Defect', 'Priority-Critical', 'OpSys-Linux'],
 'owner': {'uri': '/u/112832467119010425126/',
           'username': '5253...@gmail.com'},
 'published': '2009-10-27T16:11:09.000Z',
 'published_datetime': datetime.datetime(2009, 10, 27, 16, 11, 9),
 'stars': '0',
 'state': 'closed',
 'status': 'Fixed',
 'title': u'\u2018Bullet::BulletType\u2019 is not a class or namespace',
 'updated': '2009-10-27T19:59:41.000Z',
 'updated_datetime': datetime.datetime(2009, 10, 27, 19, 59, 41),
 'url': 'http://code.google.com/p/violetland/issues/detail?id=5'}
-- GH issue data
title: ‘Bullet::BulletType’ is not a class or namespace [moved]

This is [Issue 5](http://code.google.com/p/violetland/issues/detail?id=5) moved from a Google Code project.
Added by 2009-10-27T16:11:09.000Z by [NFLaZ...@gmail.com](http://code.google.com/u/108645236195463941083/).
Please review that bug for more context and additional comments, but update this bug.
 Closed (Fixed).

Original labels: Type-Defect, Priority-Critical, OpSys-Linux

### Original description

    [  8%] Building CXX object CMakeFiles/violetland.dir/src/game/
    Weapon.cpp.o                                                    
    /var/abs/local/yaourtbuild/violetland-svn/src/violetland/src/game/Weapon.cpp: In member 
    function ‘std::vector<Bullet*, std::allocator<Bullet*> >* Weapon::fire(float, 
    float)’:                                                                              
    /var/abs/local/yaourtbuild/violetland-svn/src/violetland/src/game/Weapon.cpp:55: ошибка: 
    ‘Bullet::BulletType’ is not a class or 
    namespace                                                                                                                   
    /var/abs/local/yaourtbuild/violetland-svn/src/violetland/src/game/Weapon.cpp:58: ошибка: 
    ‘Bullet::BulletType’ is not a class or 
    namespace                                                                                                                   
    make[2]: *** [CMakeFiles/violetland.dir/src/game/Weapon.cpp.o] Ошибка 
    1                                                       
    make[1]: *** [CMakeFiles/violetland.dir/all] Ошибка 2                                                                         
    make: *** [all] Ошибка 2                                                                                                      
    ==> ОШИБКА: Сборка завершилась неудачно.                                                                                      
        Преждевременный выход...                                                                                                  
    Error: Makepkg was unable to build violetland-svn package.

unexpected response status from Github post to create issue: 500
{'date': 'Tue, 30 Oct 2012 02:22:41 GMT', 'transfer-encoding': 'chunked', 'connection': 'keep-alive', 'status': '500', 'server': 'nginx'}

```

**Comment by trentm** [comment]
I suspect either the non-ascii quotes in the 'title' field, or the Cyrillic in the body.

**Comment by trentm** [comment]
It wasn't that at all. An attempt to add a GH issue would 500 on a body like this:

```
blah

    % blah
```

It blew up on the '%'. I'm not sure if it being in a code block matters. I suspect not.

I wasn't sending the 'Content-Type: application/json' header with the POST to the GH API. When I added that, I am able to import all of the violetland tickets.

My best guess is that some part of the GH API processing, lacking the 'Content-Type' header, attempts to parse the POST'd JSON body as application/x-www-form-urlencoded content ... and blows up finding an incomplete `%NN` percent code.

Fixed in commit 3423cd3.

Thanks for reporting this.
